runtime/src/append_vec.rs

```diff
@@ -270,6 +271,14 @@ impl<'a> Iterator for AppendVecAccountsIter<'a> {
     }
 }
 
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum MatchAccountOwnerError {
+    #[error("The account owner does not match with the provided list")]
+    NoMatch,
```

Comment:
I'm ok with this.
I was thinking though `Result<bool, MatchAccountOwnerError>`
Where bool was matched true/false
And error was only for the frustrating `UnableToLoad` case. Either way is ok with me and could be tweaked later. Making whatever you can `pub(crate)` would be great.

Reply:
but to be clear, either way is ok. This is like an enum, which is the most clear, imho.